tests/common.bash

```diff
@@ -80,10 +80,29 @@ function handle_error() {
 }
 trap 'handle_error $LINENO' ERR
 
+# A wrapper function for kubectl with retry logic
+# runs the command up to 3 times with a 5-second interval
+# to ensure successful execution
+function kubectl_retry() {
+	local max_tries=3
+	local interval=5
+	local i=0
+	while [ $i -lt $max_tries ]; do
+		kubectl $@ && return 0 || true
+		i=$((i + 1))
+		[ $i -lt $max_tries ] && echo "kubectl $@ failed, retrying in $interval seconds" || break
+		sleep $interval
+	done
+	echo "kubectl $@ failed after $max_tries tries" && return 1
+}
+
 function waitForProcess() {
 	wait_time="$1"
 	sleep_time="$2"
 	cmd="$3"
+	if [ "${cmd}" == "kubectl" ]; then
```

Comment:
I'm not sure we should enable it on `waitForProcess` as this function in practice is an auto-retry. Also `$cmd` is a complete shell line, `[ "${cmd}" == "kubectl" ]` will always return false.

For example:
```
$ cmd="kubectl apply foobar"
$ [ "${cmd}" == "kubectl" ]
$ echo $?
1
```
or am I missing something?

Reply:
I had the same impression as you @wainersm .

Here, iiuc we are mixing a retry with a wait, it might confuse users of this method. Maybe use different methods for different situations (if possible)?

Also we could be creating precedence to include other exceptions.

Reply:
Yep, let's not use `kubectl_retry` for `waitForProcess`. Thanks